### ISSUE TITLE
False positive for Nunit1029 when only a type argument is use

### ISSUE THREAD

**Issue by cbersch** [nunit/nunit.analyzers#832]
The following code
```csharp
using NUnit.Framework;

namespace NUnitTesting
{
    public sealed class GenericTestMethod
    {
        public static IEnumerable<TestCaseData> Source
            => [new TestCaseData { TypeArgs = [typeof(int)] }];

        [TestCaseSource(nameof(Source))]
        public void Generic<T>()
        {

        }
    }
}
```
raises an Nunit1029 error "The TestCaseSource provides '>0' parameter(s), but the Test methods expects '0' parameters", which shouldn't be the case.
Tested with NUnit.Analyzers version 4.6.0.

In my opinion, the check in https://github.com/nunit/nunit.analyzers/blob/master/src/nunit.analyzers/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzer.cs#L260 
```csharp
if (methodRequiredParameters + methodOptionalParameters < 1)
```
is valid only for non-generic methods and must be changed to 
```csharp
if (methodRequiredParameters + methodOptionalParameters < 1 && !testMethod.IsGenericMethod)
```
I could provide a PR for this.

**Comment by manfred-brands** [comment]
For `TestCaseData` the analyzer doesn't know how many parameters are supplied and it assumes there should at least be one.

You are right that for generic methods the count isn't correct.

You are welcome to create a PR.
